sdk/core/Azure.Core/src/ResourceType.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Structure representing a resource type.
```

Comment:
It'd be good to include here that this type is a wrapper around a "resource type" string, with a description of what resource type is in the context of ARM.